hack/README.md

```diff
@@ -0,0 +1,138 @@
+# Kubernetes hack GuideLines
+
+This document describes how you can use the scripts from [`hack`](.) directory 
+and gives a brief introduction and explanation of these scripts. 
+
+## Overview
+
+The [`hack`](.) directory contains many scripts that ensure continuous development of kubernetes, 
+enhance the robustness of the code, improve development efficiency, etc. 
+The explanations and descriptions of these scripts are helpful for contributors. 
+For details, refer to the following guidelines.
+
+## Key scripts
+
+* [`hack/boilerplate`](boilerplate): This directory contains License standard txt for various files, as well as verification scripts. 
+See [`hack/boilerplate/boilerplate.py`](boilerplate/boilerplate.py). The test directory is an example to verify boilerplate.py. 
+* [`hack/conformance`](conformance): This tool is for checking conformance e2e tests follow the requirements which is 
+https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#conformance-test-requirements. 
+* [`hack/e2e-internal`](e2e-internal): 
+* [`hack/gen-swagger-doc`](gen-swagger-doc): This script generates docs from the latest swagger spec. 
+* [`hack/jenkins`](jenkins): Jenkins is no longer in use, please see the [test-infra repo](https://github.com/kubernetes/test-infra) 
+and in particular [Prow](https://github.com/kubernetes/test-infra/tree/master/prow) instead.
+* [`hack/lib`](lib): This directory contains some common scripts, such as `etcd.sh`, `golang.sh`, `init.sh`, `logging.sh`, `protoc.sh`, `swagger.sh`, `test.sh`, `util.sh`, `version.sh`. 
+* [`hack/make-rules`](make-rules): 
+* [`hack/testdata`](testdata): The directory contains many test data, such as yaml files, json files, tls.crt, tls.key, etc. 
+* [`hack/verify-flags`](verify-flags): This `excluded-flags.txt` file contains some flags used by `hack/verify-flags-underscore.py`. 
+* [`.golint_failures`](.golint_failures): This file contains the result after running `hack/verify-golint.sh`.
```

Comment:
This seems like something that should go in verify-golint.sh, same for the other helper files